src/main/java/dyorgio/runtime/macos/trayicon/fixer/jna/foundation/NSObject.java

```diff
@@ -39,28 +39,33 @@
     protected static final Pointer allocSel = Foundation.INSTANCE.sel_registerName("alloc");
     protected static final Pointer initSel = Foundation.INSTANCE.sel_registerName("init");
     protected static final Pointer releaseSel = Foundation.INSTANCE.sel_registerName("release");
+    protected static final Pointer retainSel = Foundation.INSTANCE.sel_registerName("retain");
     protected static final Pointer performSelectorOnMainThread$withObject$waitUntilDoneSel
             = Foundation.INSTANCE.sel_registerName("performSelectorOnMainThread:withObject:waitUntilDone:");
 
     final NativeLong id;
 
     public NSObject(NativeLong id) {
         this.id = id;
+        Foundation.INSTANCE.objc_msgSend(id, retainSel);
```

Comment:
Not ok assume that all child/instances of NSObject will be "retained".
Some objects are created as autorelease per example.